docs/typescript.md

```diff
@@ -54,168 +68,147 @@ fn main(x: u32, y: pub u32) -> pub u32 {
 
 ```ts
-export const getAcir = async () => {
-  const { circuit, abi } = await compileCircuit();
-  await initialiseAztecBackend();
+import {
+  Crs,
+  newBarretenbergApiAsync,
+  RawBuffer,
+} from '@aztec/bb.js/dest/node';
+
+const api = await newBarretenbergApiAsync(4);
+
+const [exact, total, subgroup] = await api.acirGetCircuitSizes(JSON.parse(circuit.bytecode));
+const subgroupSize = Math.pow(2, Math.ceil(Math.log2(circuitSize)));
+const crs = await Crs.new(subgroupSize + 1);
+await api.commonInitSlabAllocator(subgroupSize);
+await api.srsInitSrs(
+  new RawBuffer(crs.getG1Data()),
+  crs.numPoints,
+  new RawBuffer(crs.getG2Data()),
+);
+
+const acirComposer = await api.acirNewAcirComposer(subgroupSize);
 
-  let acir_bytes = new Uint8Array(Buffer.from(circuit, 'hex'));
-  return acir_read_bytes(acir_bytes);
-};
 ```
 
-Calling `getAcir()` now should return us the ACIR of the circuit, ready to be used in proofs.
+We should take two very useful objects from here: `api` and `acirComposer`. Make sure to keep these close by!
 
-## Initializing Prover & Verifier
+:::info
 
-Prior to proving and verifying, the prover and verifier have to first be initialized by calling
-`barretenberg`'s `setup_generic_prover_and_verifier` with your Noir program's ACIR:
+On the browser, you also need to init the ACVM. You can do that by importing it and calling it like:
 
 ```ts
-let [prover, verifier] = await setup_generic_prover_and_verifier(acir);
+import initACVM, { executeCircuit, compressWitness } from '@noir-lang/acvm_js';
+
+await initACVM();
+// the rest of your code
 ```
 
-This is probably a good time to store this prover and verifier into your state like React Context,
-Redux, or others.
+:::
 
-## Proving
+## Generating witnesses
 
-The Noir program can then be executed and proved by calling `barretenberg`'s `create_proof`
-function:
+Witness generation is what allows us to prove with arbitrary inputs (like user inputs on a form, game, etc). In this example, our input is a simple object with our circuit inputs `x`, `y`, and return `z` (fun fact: the return value in Noir is actually a public input!). We're wrapping it in a function, so it can be conveniently called later on.
 
 ```ts
-const proof = await create_proof(prover, acir, abi);
-```
+import { ethers } from 'ethers'; // I'm lazy so I'm using ethers to pad my input
+import { executeCircuit, compressWitness } from '@noir-lang/acvm_js';
 
-On the browser, this proof can fail as it requires heavy loads to be run on worker threads. Here's a
-quick example of a worker:
+async function generateWitness(input: any, acirBuffer: Buffer): Promise<Uint8Array> {
+  const initialWitness = new Map<number, string>();
+  initialWitness.set(1, ethers.utils.hexZeroPad(`0x${input.x.toString(16)}`, 32));
+  initialWitness.set(2, ethers.utils.hexZeroPad(`0x${input.y.toString(16)}`, 32));
+  initialWitness.set(3, ethers.utils.hexZeroPad(`0x${input.z.toString(16)}`, 32));
```

Comment:
including this line causes an error for me. If i comment it out, everything works as expected.